packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcherTrigger.tsx

```diff
@@ -1,55 +1,66 @@
-  (props, ref) => {
-    const user = useCoreUser();
-    const { organization } = useCoreOrganization();
-    const { hidePersonal } = useOrganizationSwitcherContext();
+const _OrganizationSwitcherTrigger = withAvatarShimmer((props: OrganizationSwitcherTriggerProps) => {
+  const { sx, buttonRef, ...rest } = props;
+  const user = useCoreUser();
+  const { organization } = useCoreOrganization();
+  const { hidePersonal } = useOrganizationSwitcherContext();
 
-    return (
-      <Button
-        elementDescriptor={descriptors.organizationSwitcherTrigger}
-        variant='ghost'
-        colorScheme='neutral'
-        sx={t => ({ minHeight: 0, padding: `0 ${t.space.$2} 0 0` })}
-        {...props}
-        ref={ref}
-      >
-        {organization && (
-          <OrganizationPreview
-            elementId={'organizationSwitcher'}
-            gap={3}
-            size={'sm'}
-            organization={organization}
-            sx={{ maxWidth: '30ch' }}
-          />
-        )}
-        {!organization && (
-          <PersonalWorkspacePreview
-            size={'sm'}
-            gap={3}
-            user={{ profileImageUrl: user.profileImageUrl }}
-            title={
-              hidePersonal
-                ? localizationKeys('organizationSwitcher.notSelected')
-                : localizationKeys('organizationSwitcher.personalWorkspace')
-            }
-          />
-        )}
-        <Icon
-          elementDescriptor={descriptors.organizationSwitcherTriggerIcon}
-          icon={Selector}
-          sx={t => ({ opacity: t.opacity.$sm, marginLeft: `${t.space.$2}` })}
+  return (
+    <Button
+      elementDescriptor={descriptors.organizationSwitcherTrigger}
+      variant='ghost'
+      colorScheme='neutral'
+      sx={[t => ({ minHeight: 0, padding: `0 ${t.space.$2} 0 0` }), sx]}
+      {...rest}
+      ref={buttonRef}
+    >
+      {organization && (
+        <OrganizationPreview
+          elementId={'organizationSwitcher'}
+          gap={3}
+          size={'sm'}
+          organization={organization}
+          sx={{ maxWidth: '30ch' }}
         />
-      </Button>
-    );
-  },
+      )}
+      {!organization && (
+        <PersonalWorkspacePreview
+          size={'sm'}
+          gap={3}
+          user={{ profileImageUrl: user.profileImageUrl }}
+          title={
+            hidePersonal
+              ? localizationKeys('organizationSwitcher.notSelected')
+              : localizationKeys('organizationSwitcher.personalWorkspace')
+          }
+        />
+      )}
+      <Icon
+        elementDescriptor={descriptors.organizationSwitcherTriggerIcon}
+        icon={Selector}
+        sx={t => ({ opacity: t.opacity.$sm, marginLeft: `${t.space.$2}` })}
+      />
+    </Button>
+  );
+});
+
+export const OrganizationSwitcherTrigger = forwardRef<HTMLButtonElement, OrganizationSwitcherTriggerProps>(
+  (props, ref) => (
+    <_OrganizationSwitcherTrigger
```

Comment:
Just double checking - does the ref work even through the avatar shimmer HOC?

Reply:
Yes! Otherwise the popover would not open.